File: app/src/main/java/com/example/movies/ui/search/view/SearchFragment.java
package com.example.movies.ui.search.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.view.View;

import com.example.movies.R;
import com.example.movies.core.base.BaseFragment;
import com.example.movies.ui.search.adapter.TabLayoutAdapterSearch;

import butterknife.BindView;
import butterknife.Unbinder;


/**
 * Created by Tarik on 16.11.2019.
 */

public class SearchFragment extends BaseFragment {

    @BindView(R.id.tab_layout_search)
    TabLayout tabLayoutSearch;
    @BindView(R.id.search_view_pager)
    ViewPager searchViewPager;

    @Override
    public Integer getFragmentLayoutId() {
        return R.layout.search_fragment_layout;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        TabLayoutAdapterSearch tabLayoutAdapterSearch = new TabLayoutAdapterSearch(getActivity()
                .getSupportFragmentManager(), getContext());
        searchViewPager.setAdapter(tabLayoutAdapterSearch);
        tabLayoutSearch.setupWithViewPager(searchViewPager);
    }
}

File: app/src/main/java/com/example/movies/ui/search/view/SearchFragment.kt
package com.example.movies.ui.search.view

import android.os.Bundle
import android.view.View
import com.example.movies.R
import com.example.movies.core.base.BaseFragment
import com.example.movies.ui.search.adapter.TabLayoutAdapterSearch
import kotlinx.android.synthetic.main.search_fragment_layout.*

class SearchFragment : BaseFragment() {
    override fun getFragmentLayoutId(): Int = R.layout.search_fragment_layout

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val tabLayoutAdapterSearch: TabLayoutAdapterSearch = TabLayoutAdapterSearch(activity!!.supportFragmentManager, context!!)
        search_view_pager.adapter = tabLayoutAdapterSearch
        tab_layout_search.setupWithViewPager(search_view_pager)
    }
}